Memeolist/Memeolist/AppDelegate.swift

```diff
@@ -11,27 +11,36 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     var window: UIWindow?
     var profile: ProfileDetails!
 
+    // Redirects the user to the login screen
+    func requireLogin() -> Bool {
+        let rootViewController = LoginViewController()
+        window?.rootViewController = UINavigationController(rootViewController: rootViewController)
+        window?.makeKeyAndVisible()
+        return true
+    }
+
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
         IQKeyboardManager.shared.enable = true
 
         // create the authentication config
         let authenticationConfig = AuthenticationConfig(redirectURL: "memeolist://callback")
-        try! AgsAuth.instance.configure(authConfig: authenticationConfig)
-        if let transport = AgsSync.instance.transport {
-            transport.headerProvider = AgsAuth.instance.getAuthHeaderProvider()
-        }
-        
-        guard let user = try! AgsAuth.instance.currentUser() else {
-            let rootViewController = LoginViewController()
-            window?.rootViewController = UINavigationController(rootViewController: rootViewController)
-            window?.makeKeyAndVisible()
-            return true
+        do {
+            try AgsAuth.instance.configure(authConfig: authenticationConfig)
+            if let transport = AgsSync.instance.transport {
+                transport.headerProvider = AgsAuth.instance.getAuthHeaderProvider()
+            }
+
+            guard let _ = try! AgsAuth.instance.currentUser() else {
```

Comment:
I'm affaid that guard is not going to work here with `try!`. We should change this to `try?` and it will feed the purpose.